hydroflow_lang/src/graph/ops/fold.rs

```diff
@@ -81,13 +80,12 @@ pub const FOLD: OperatorConstraints = OperatorConstraints {
         let iterator_item_ident = wc.make_ident("iterator_item");
 
         let (write_prologue, write_iterator, write_iterator_after) = match persistence {
-            // TODO(mingwei): Issues if initial value is not copy.
-            // TODO(mingwei): Might introduce the initial value multiple times on scheduling.
             Persistence::Tick => (
                 Default::default(),
                 quote_spanned! {op_span=>
                     let #ident = {
-                        let mut #accumulator_ident = #init;
+                        #[allow(clippy::redundant_closure_call)]
+                        let mut #accumulator_ident = (#init)();
```

Comment:
May be worth trying to store `#init` in its own name ident in the prologue, or put a TODO here stating to try that again

Reply:
what would that give us?